pkg/controller/common/reconciler.go

```diff
@@ -198,15 +203,47 @@ func (r *BaseIngressReconciler) reconcileRoute(ctx context.Context, ci networkin
 }
 
 func (r *BaseIngressReconciler) reconcileDeletion(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
-	// TODO: something with a finalizer?  We're using owner refs for
-	// now, but really shouldn't be using owner refs from
-	// cluster-scoped ClusterIngress to a namespace-scoped K8s
-	// Service.
-	return nil
+	// get list of ingress object for a namespace
+	ingressList := networkingv1alpha1.IngressList{}
+	if err := r.Client.List(ctx, &client.ListOptions{
+		Namespace: ci.GetNamespace(),
+	}, &ingressList); err != nil {
+		return err
+	}
+	// If particular namespace has only one ingress object then after deletion namespace will have empty ingress object
+	// So remove namespace from SMMR
+	if len(ingressList.Items) == 1 {
+		for i := range ingressList.Items {
+			// In order to double check that we are reconciling proper ingress check with name and namespace
```

Comment:
The for loop is unnecessary. You know it only contains 1 item so you can reference [0] immediately